Web Inspector: Introduce the Frame target type to prep for site isolation

the-chenergy opened the pull request:
#### 06f8ad1a5a66f9ffaa33696a5b9fba4f4c65070b
<pre>
Web Inspector: Introduce the Frame target type to prep for site isolation
<a href="https://rdar.apple.com/143599045">rdar://143599045</a>
bugs.webkit.org/show_bug.cgi?id=298742

Reviewed by BJ Burg.

This patch introduces the frame target type for Web Inspector. A target
represents a object that the inspector inspects, each carrying its own
resources and providing agents, groups of commands and events by domains
of functionalities. For a long time, there exists one page target per
web page, which worked fine with a single web process per page. Now with
site isolation being introduced, a better design is to treat each
individual frame in the page as an inspection unit, so we&apos;re not relying
on them all sharing the same web process to work.

The frame target is intended to eventually be the contact point for most
Web Inspector domains that were originally owned by the page target.
This patch is a step in that direction by introducing the frame target
type and merely making it possible to co-exist with other targets in
the frontend. Apart from existing, the frame target currently has no
domains (not even a Target domain) and relies on its page&apos;s Target
domain to notify its lifecycle events. The plan is to change that while
bringing in other domains in future commits.

The contents of this patch break down into three interlinked components:
1. Define the frame-related objects: FrameInspectorController, WebFrameInspectorTarget,
   and WebFrameInspectorTargetProxy in the backend, and FrameTarget in
   the frontend.
2. Relax parts of the frontend code to work with FrameTarget that has
   missing agents/domains, while updating relevant tests.
3. Add a test case for frames&apos; lifecycle events linked to the creation
   and deletion of frame targets, also as a prototype of inspector tests
   with site isolation.

To summarize the structure of the frame-related objects, in comparison
with the existing page-related objects:
* In WebCore:
  - (old) Page owns InspectorController, which owns the agents
  - (new) Frame owns FrameInspectorController, which will own agents
    that we&apos;ll slowly create or move from the page
* In WebKit, web process:
  - (old) WebPage owns WebPageInspectorTarget through WebPageInspectorTargetController
  - (new) WebFrame owns WebFrameInspectorTarget directly, given we don&apos;t
    think there will be a need for a controller with just one target
    per frame
* In WebKit, UI process:
  - (old) WebPageProxy owns InspectorTargetProxy through WebPageInspectorController,
    which owns the Target agent
  - (new) We rename InspectorTargetProxy to WebPageInspectorTargetProxy
    to more easily introduce a frame-equivalent
  - (new) WebFrameProxy reuses the WebPageInspectorController to manage
    WebFrameInspectorTargetProxy

Test: http/tests/site-isolation/inspector/target/target.html
* LayoutTests/http/tests/inspector/target/provisional-load-cancels-previous-load.html:
* LayoutTests/http/tests/inspector/target/target-events-for-provisional-page.html:
* LayoutTests/http/tests/site-isolation/inspector/target/resources/leaf-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/inspector/target/resources/middle-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/inspector/target/target-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/inspector/target/target.html: Added.
* LayoutTests/inspector/unit-tests/target-manager-expected.txt:
* LayoutTests/inspector/unit-tests/target-manager.html:
* LayoutTests/inspector/worker/debugger-shared-breakpoint.html:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/JavaScriptCore/inspector/InspectorTarget.h:
* Source/JavaScriptCore/inspector/agents/InspectorTargetAgent.cpp:
(Inspector::targetTypeToProtocolType):
* Source/JavaScriptCore/inspector/protocol/Target.json:
* Source/JavaScriptCore/inspector/scripts/codegen/models.py:
(validate_target_types):
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/FrameInspectorController.cpp: Added.
(WebCore::FrameInspectorController::FrameInspectorController):
(WebCore::FrameInspectorController::~FrameInspectorController):
(WebCore::FrameInspectorController::ref const):
(WebCore::FrameInspectorController::deref const):
(WebCore::FrameInspectorController::createLazyAgents):
(WebCore::FrameInspectorController::connectFrontend):
(WebCore::FrameInspectorController::disconnectFrontend):
(WebCore::FrameInspectorController::dispatchMessageFromFrontend):
(WebCore::FrameInspectorController::developerExtrasEnabled const):
(WebCore::FrameInspectorController::canAccessInspectedScriptState const):
(WebCore::FrameInspectorController::functionCallHandler const):
(WebCore::FrameInspectorController::evaluateHandler const):
(WebCore::FrameInspectorController::frontendInitialized):
(WebCore::FrameInspectorController::executionStopwatch const):
(WebCore::FrameInspectorController::debugger):
(WebCore::FrameInspectorController::vm):
* Source/WebCore/inspector/FrameInspectorController.h: Added.
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::Frame):
(WebCore::Frame::protectedInspectorController):
* Source/WebCore/page/Frame.h:
(WebCore::Frame::inspectorController):
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Base/Main.js:
* Source/WebInspectorUI/UserInterface/Base/TargetType.js:
* Source/WebInspectorUI/UserInterface/Controllers/CanvasManager.js:
(WI.CanvasManager.prototype.setRecordingAutoCaptureFrameCount):
* Source/WebInspectorUI/UserInterface/Controllers/ConsoleManager.js:
(WI.ConsoleManager.prototype.initializeTarget):
(WI.ConsoleManager.prototype.requestClearMessages):
* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager.prototype.addEventBreakpoint):
(WI.DOMDebuggerManager.prototype.addURLBreakpoint):
(WI.DOMDebuggerManager.prototype._handleEventBreakpointDisabledStateChanged):
(WI.DOMDebuggerManager.prototype._handleEventBreakpointEditablePropertyChanged):
(WI.DOMDebuggerManager.prototype._handleURLBreakpointDisabledStateChanged):
(WI.DOMDebuggerManager.prototype._handleURLBreakpointEditablePropertyChanged):
(WI.DOMDebuggerManager.prototype.allSupportedTargets):
(WI.DOMDebuggerManager):
* Source/WebInspectorUI/UserInterface/Controllers/DOMManager.js:
(WI.DOMManager.prototype._handleEventBreakpointEditablePropertyChanged):
* Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js:
(WI.DebuggerManager.prototype.async initializeTarget):
(WI.DebuggerManager.prototype.set breakpointsEnabled):
(WI.DebuggerManager.prototype.setShouldBlackboxScript):
(WI.DebuggerManager.prototype.setShouldBlackboxPattern):
(WI.DebuggerManager.prototype.set asyncStackTraceDepth):
(WI.DebuggerManager.prototype.addSymbolicBreakpoint):
(WI.DebuggerManager.prototype._setBreakpoint):
(WI.DebuggerManager.prototype._removeBreakpoint):
(WI.DebuggerManager.prototype._updateSpecialBreakpoint):
(WI.DebuggerManager.prototype._handleSymbolicBreakpointDisabledStateChanged):
(WI.DebuggerManager.prototype._handleSymbolicBreakpointEditablePropertyChanged):
(WI.DebuggerManager.prototype._handleBlackboxBreakpointEvaluationsChange):
(WI.DebuggerManager.prototype._handleEngineeringPauseForInternalScriptsSettingChanged):
(WI.DebuggerManager.prototype._handleSourceCodeSourceMapAdded):
(WI.DebuggerManager.prototype.allSupportedTargets):
(WI.DebuggerManager):
* Source/WebInspectorUI/UserInterface/Controllers/NetworkManager.js:
(WI.NetworkManager.prototype.set emulatedCondition):
* Source/WebInspectorUI/UserInterface/Controllers/RuntimeManager.js:
(WI.RuntimeManager.prototype.initializeTarget):
* Source/WebInspectorUI/UserInterface/Controllers/TargetManager.js:
(WI.TargetManager.prototype._createTarget):
* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Models/HeapAllocationsInstrument.js:
(WI.HeapAllocationsInstrument.prototype.startInstrumentation):
(WI.HeapAllocationsInstrument.prototype.stopInstrumentation):
(WI.HeapAllocationsInstrument.prototype._takeHeapSnapshot):
(WI.HeapAllocationsInstrument.prototype.allSupportedTargets):
(WI.HeapAllocationsInstrument):
* Source/WebInspectorUI/UserInterface/Protocol/FrameTarget.js: Copied from Source/WebInspectorUI/UserInterface/Base/TargetType.js.
(WI.FrameTarget):
* Source/WebInspectorUI/UserInterface/Protocol/Target.js:
(WI.Target.prototype.initialize):
* Source/WebInspectorUI/UserInterface/Test.html:
* Source/WebInspectorUI/UserInterface/Views/HeapAllocationsTimelineView.js:
(WI.HeapAllocationsTimelineView.prototype._takeHeapSnapshotClicked):
* Source/WebInspectorUI/UserInterface/Views/LogContentView.js:
(WI.LogContentView.prototype._garbageCollect):
* Source/WebInspectorUI/UserInterface/Views/ScopeChainDetailsSidebarPanel.js:
* Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
(WI.SettingsTabContentView.prototype._createConsoleSettingsView):
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.cpp:
(WebKit::InspectorTargetProxy::InspectorTargetProxy):
(WebKit::InspectorTargetProxy::create): Deleted.
(WebKit::InspectorTargetProxy::connect): Deleted.
(WebKit::InspectorTargetProxy::disconnect): Deleted.
(WebKit::InspectorTargetProxy::sendMessageToTargetBackend): Deleted.
(WebKit::InspectorTargetProxy::didCommitProvisionalTarget): Deleted.
(WebKit::InspectorTargetProxy::isProvisional const): Deleted.
* Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h:
* Source/WebKit/UIProcess/Inspector/WebFrameInspectorTargetProxy.cpp: Added.
(WebKit::WebFrameInspectorTargetProxy::create):
(WebKit::WebFrameInspectorTargetProxy::WebFrameInspectorTargetProxy):
(WebKit::WebFrameInspectorTargetProxy::connect):
(WebKit::WebFrameInspectorTargetProxy::disconnect):
(WebKit::WebFrameInspectorTargetProxy::sendMessageToTargetBackend):
(WebKit::WebFrameInspectorTargetProxy::didCommitProvisionalTarget):
(WebKit::WebFrameInspectorTargetProxy::isProvisional const):
* Source/WebKit/UIProcess/Inspector/WebFrameInspectorTargetProxy.h: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h.
(isType):
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
(WebKit::WebPageInspectorController::init):
(WebKit::WebPageInspectorController::createWebPageInspectorTarget):
(WebKit::WebPageInspectorController::createWebFrameInspectorTarget):
(WebKit::WebPageInspectorController::didCreateProvisionalPage):
(WebKit::WebPageInspectorController::createInspectorTarget): Deleted.
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.h:
* Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.cpp: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.cpp.
(WebKit::WebPageInspectorTargetProxy::create):
(WebKit::WebPageInspectorTargetProxy::WebPageInspectorTargetProxy):
(WebKit::WebPageInspectorTargetProxy::connect):
(WebKit::WebPageInspectorTargetProxy::disconnect):
(WebKit::WebPageInspectorTargetProxy::sendMessageToTargetBackend):
(WebKit::WebPageInspectorTargetProxy::didCommitProvisionalTarget):
(WebKit::WebPageInspectorTargetProxy::isProvisional const):
* Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.h: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h.
(isType):
* Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
(WebKit::ProvisionalFrameProxy::protectedFrame const):
* Source/WebKit/UIProcess/ProvisionalFrameProxy.h:
(WebKit::ProvisionalFrameProxy::frame const):
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockHidService.cpp:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::WebFrameProxy):
(WebKit::WebFrameProxy::~WebFrameProxy):
(WebKit::WebFrameProxy::takeProvisionalFrame):
(WebKit::WebFrameProxy::provisionalLoadProcess):
(WebKit::WebFrameProxy::prepareForProvisionalLoadInProcess):
(WebKit::WebFrameProxy::sendMessageToInspectorFrontend):
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::~WebPageProxy):
(WebKit::WebPageProxy::setFindClient):
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::launchProcessForReload):
(WebKit::WebPageProxy::reload):
(WebKit::WebPageProxy::createInspectorTarget):
(WebKit::WebPageProxy::viewWillStartLiveResize):
(WebKit::WebPageProxy::wheelEventHandlingCompleted):
(WebKit::WebPageProxy::receivedPolicyDecision):
(WebKit::WebPageProxy::listenForLayoutMilestones):
(WebKit::WebPageProxy::didFailProvisionalLoadForFrame):
(WebKit::WebPageProxy::didReceiveTitleForFrame):
(WebKit::WebPageProxy::nonEphemeralWebPageProxy):
(WebKit::WebPageProxy::logFrameNavigation):
(WebKit::WebPageProxy::decidePolicyForNewWindowAction):
(WebKit::WebPageProxy::didPerformClientRedirectShared):
(WebKit::WebPageProxy::didPerformServerRedirectShared):
(WebKit::WebPageProxy::setMediaVolume):
(WebKit::WebPageProxy::Internals::currentlyProcessedMouseDownEvent):
(WebKit::WebPageProxy::contextMenuItemSelected):
(WebKit::WebPageProxy::updateEditorState):
(WebKit::WebPageProxy::processDidBecomeResponsive):
(WebKit::WebPageProxy::setScrollPinningBehavior):
(WebKit::WebPageProxy::resetSpeechSynthesizer):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTarget.cpp: Added.
(WebKit::WebFrameInspectorTarget::WebFrameInspectorTarget):
(WebKit::WebFrameInspectorTarget::protectedFrame):
(WebKit::WebFrameInspectorTarget::identifier const):
(WebKit::WebFrameInspectorTarget::connect):
(WebKit::WebFrameInspectorTarget::disconnect):
(WebKit::WebFrameInspectorTarget::sendMessageToTargetBackend):
(WebKit::WebFrameInspectorTarget::toTargetID):
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTarget.h: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h.
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTargetFrontendChannel.cpp: Copied from Source/WebKit/UIProcess/ProvisionalFrameProxy.h.
(WebKit::WebFrameInspectorTargetFrontendChannel::WebFrameInspectorTargetFrontendChannel):
(WebKit::WebFrameInspectorTargetFrontendChannel::protectedFrame):
(WebKit::WebFrameInspectorTargetFrontendChannel::sendMessageToFrontend):
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTargetFrontendChannel.h: Copied from Source/WebKit/UIProcess/ProvisionalFrameProxy.h.
* Source/WebKit/WebProcess/UserContent/WebUserContentController.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidCommitLoad):
* Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp:
(WebKit::WebPermissionController::query):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::WebFrame):
(WebKit::WebFrame::connectInspector):
(WebKit::WebFrame::disconnectInspector):
(WebKit::WebFrame::sendMessageToInspectorTarget):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebFrame.messages.in:

Canonical link: <a href="https://commits.webkit.org/300338@main">https://commits.webkit.org/300338@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8c906311750eb5cf304ba3acc4d75c7bd5c705b7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/122000 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/41702 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/32372 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/128560 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/74093 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/77661959-ae42-4d9a-a3e0-ccd17d426fbc) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/123876 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/42417 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/50296 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/92719 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/61603 "Passed tests") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/c6f4e8f5-e776-449c-928a-261660253262) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/124952 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/33989 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/109252 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/73394 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/0443f213-6a07-4a33-9261-878f0b744981) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/32832 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/27413 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/72057 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/114148 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/103327 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/27609 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/131324 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/120526 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/48939 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/37212 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/101279 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/49313 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/105466 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/101150 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25685 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/46519 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/24636 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/45640 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/48796 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/54530 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/150705 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/48266 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/38556 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/51616 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/49946 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

burg commented on Source/WebCore/page/Frame.cpp:
```diff
@@ -119,6 +120,7 @@ Frame::Frame(Page& page, FrameIdentifier frameID, FrameType frameType, HTMLFrame
     , m_navigationScheduler(makeUniqueRefWithoutRefCountedCheck<NavigationScheduler>(*this))
     , m_opener(opener)
     , m_frameTreeSyncData(WTFMove(frameTreeSyncData))
+    , m_inspectorController(makeUniqueRefWithoutRefCountedCheck<FrameInspectorController>(*this))
```
After this refactor I think [Web]PageInspectorController would logically be for agents implemented in UIProcess?

the-chenergy replied:
Right, the plan is to eventually move the agents required to be managed by a page to the UI process instead, so `InspectorController` (possibly renamed to `PageInspectorController` later) may still exist and tied to the page still.

the-chenergy replied:
It was mainly to make the Safer-C++ bot happy as it keeps pointing out stuff like this is not safe:
```cpp
if (m_provisionalFrame)
    m_provisionalFrame->...
```
Which is what I otherwise would have in `WebFrameInspectorTargetProxy.cpp` in a bunch of places.

burg commented on LayoutTests/http/tests/site-isolation/inspector/target/target.html:
```diff
@@ -0,0 +1,35 @@
+                    let iframe = document.createElement("iframe");
+                    iframe.src = "http://127.0.0.1:8000/site-isolation/inspector/target/resources/middle-iframe.html";
+                    document.body.appendChild(iframe);
+                    setTimeout(function () {
```
Nit: could be an arrow function to shorten.

burg commented on LayoutTests/http/tests/site-isolation/inspector/target/target.html:
```diff
@@ -0,0 +1,35 @@
+<script src="../../../inspector/resources/inspector-test.js"></script>
+<script>
+    function test() {
+        let suite = InspectorTest.createAsyncSuite("site-isolation.inspector.target");
```
Style: try to match existing convention for suite names: `Domain.Command.Suitename` and for test cases:
`Domain.Command.Suitename.Testname`

burg commented on LayoutTests/http/tests/site-isolation/inspector/target/target.html:
```diff
@@ -0,0 +1,35 @@
+                    document.body.appendChild(iframe);
+                    setTimeout(function () {
+                        document.body.removeChild(iframe);
+                    }, 100);
```
Recommend against using hardcoded timeouts. I'd recommend having one `eval` to set up the page contents, then a second `eval` to actually trigger the behavior under test.

burg commented on Source/WebCore/inspector/FrameInspectorController.cpp:
```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (C) 2011 Google Inc. All rights reserved.
+ * Copyright (C) 2014, 2015 Apple Inc. All rights reserved.
```
Nit; bump copyright

the-chenergy commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.cpp:
```diff
@@ -0,0 +1,117 @@
+
+    Ref page = m_page.get();
+    if (page->hasRunningProcess())
+        page->protectedLegacyMainFrameProcess()->send(Messages::WebPage::ConnectInspector(identifier(), connectionType), page->webPageIDInMainFrameProcess());
```
I think not easily until we move the page target to be controlled from the UI process instead.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/dfb1a01c3dc5c8535555746d8b28a8cabebd5266. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/0dc88b604684d3005f8635f68fff919b69ef7335. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/275640b5fc9bf988c0e47101fa5eb612f58be06d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/32fc157ed76168e08d9eee0a929f12b8615c0b1b. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

burg commented on Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
```diff
@@ -1008,6 +1008,14 @@ WI.DOMDebuggerManager = class DOMDebuggerManager extends WI.Object
     }
+
+    // FIXME <https://webkit.org/b/298981> Add DOMDebugger support for FrameTarget.
+    *_allSupportedTargets() {
```
Ooh, cool new generator syntax for me to figure out :P

Though I suppose fully modern syntax would be 

```
*#allSupportedTargets() {
```

burg commented on Source/WebInspectorUI/UserInterface/Controllers/DOMManager.js:
```diff
@@ -875,6 +875,10 @@ WI.DOMManager = class DOMManager extends WI.Object
             return;
 
         for (let target of WI.targets) {
```
Should we switch this one as well?

the-chenergy replied:
Seems like this is only done once in this file so I'll keep it for now.

burg commented on Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js:
```diff
@@ -615,15 +619,15 @@ WI.DebuggerManager = class DebuggerManager extends WI.Object
                 target.DebuggerAgent.setShouldBlackboxURL(regex.source, !!shouldBlackbox, !regex.ignoreCase, isRegex);
         }
 
         for (let sourceMap of WI.SourceMap.instances) {
```
It's unclear if we should attempt to group source maps by target, so this is fine for now. We'll need to fine-tune such things later.

burg commented on Source/WebKit/Sources.txt:
```diff
@@ -612,6 +612,8 @@ UIProcess/GPU/GPUProcessProxy.cpp
 UIProcess/Model/ModelProcessProxy.cpp
 
 UIProcess/Inspector/InspectorTargetProxy.cpp
+UIProcess/Inspector/WebPageInspectorTargetProxy.cpp
```
Nit: alphabetization

burg commented on Source/WebKit/UIProcess/Inspector/WebPageInspectorController.h:
```diff
@@ -26,9 +26,12 @@
 #include <JavaScriptCore/InspectorTargetAgent.h>
+#include <WebCore/FrameIdentifier.h>
 #include <WebCore/PageIdentifier.h>
+#include <memory>
```
why is <memory> needed here?

burg commented on Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.cpp:
```diff
@@ -33,84 +33,17 @@
 #include "WebPageMessages.h"
 #include "WebPageProxy.h"
 #include "WebProcessProxy.h"
+#include <JavaScriptCore/InspectorTarget.h>
```
Nit: TZoneMallocInlines.h no longer needed?

burg commented on Source/WebKit/Sources.txt:
```diff
@@ -726,6 +728,8 @@ WebProcess/Inspector/WebInspectorUIExtensionController.cpp
 WebProcess/Inspector/WebPageInspectorTarget.cpp
 WebProcess/Inspector/WebPageInspectorTargetController.cpp
 WebProcess/Inspector/WebPageInspectorTargetFrontendChannel.cpp
+WebProcess/Inspector/WebFrameInspectorTarget.cpp
```
Nit: alpha

burg commented on Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
```diff
@@ -178,6 +179,7 @@ Ref<WebFrame> WebFrame::createRemoteSubframe(WebPage& page, WebFrame& parent, We
 WebFrame::WebFrame(WebPage& page, WebCore::FrameIdentifier frameID)
     : m_page(page)
     , m_frameID(frameID)
+    , m_inspectorTarget(makeUniqueRef<WebFrameInspectorTarget>(*this))
```
is there any downside to doing this initialization lazily and using a RefPtr? I'm concerned about the overhead for the vast majority of users who aren't inspecting.

burg commented on Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp:
```diff
@@ -70,7 +70,7 @@ WebPermissionController::~WebPermissionController()
 void WebPermissionController::query(WebCore::ClientOrigin&& origin, WebCore::PermissionDescriptor descriptor, const WeakPtr<WebCore::Page>& page, WebCore::PermissionQuerySource source, CompletionHandler<void(std::optional<WebCore::PermissionState>)>&& completionHandler)
 {
 #if ENABLE(WEB_PUSH_NOTIFICATIONS)
-    if (WebCore::DeprecatedGlobalSettings::builtInNotificationsEnabled() && (descriptor.name == PermissionName::Notifications || descriptor.name == PermissionName::Push)) {
```
is this part of the change needed?

the-chenergy replied:
I run into build problems due to unified build changes.

burg commented on Source/WebKit/UIProcess/WebAuthentication/Mock/MockHidService.cpp:
```diff
@@ -28,6 +28,7 @@
 
 #if ENABLE(WEB_AUTHN)
 
+#include "Logging.h"
```
Why?

the-chenergy replied:
Due to unified build changes, probably.

burg commented on Source/WebKit/UIProcess/Inspector/WebFrameInspectorTargetProxy.cpp:
```diff
@@ -0,0 +1,111 @@
+
+void WebFrameInspectorTargetProxy::sendMessageToTargetBackend(const String& message)
+{
+    if (RefPtr provisionalFrame = m_provisionalFrame.get()) {
```
Okay, I have a design question. is there any relationship between m_frame and m_provisionalFrame, or are they being used mutually exclusively? I know that this mirrors the PageTarget code that already exists, I'm just not sure if we are calling disconnect() on the provisional frame target before the m_provisionalFrame gets nulled out.

Alternatively, can you add a bugzilla link to track adding more tests for provisional frame targets? I think my questions would be answered by the test case.

burg reviewed (approved):
r=me. please ensure layout tests run successfully on EWS on your final PR before merging.

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/4b23dbd7de8cfdd98aa9c364d54c6b937e09baac. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

the-chenergy commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -908,7 +908,7 @@ WebPageProxy::WebPageProxy(PageClient& pageClient, WebProcessProxy& process, Ref
 #if PLATFORM(IOS_FAMILY)
     DeprecatedGlobalSettings::setDisableScreenSizeOverride(m_preferences->disableScreenSizeOverride());
-    
+
```
Unnecessary changes. (Ugh.)

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/a1e56ad4af6b9c38c93353573caceb1d47156069. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ba3954c00fc66950d3cedc91836ee04f6dfd5bdd. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

the-chenergy commented on LayoutTests/platform/mac-wk1/TestExpectations:
```diff
@@ -1780,6 +1780,10 @@ inspector/heap/getRemoteObject.html [ Skip ]
 inspector/unit-tests/heap-snapshot-collection-event.html [ Skip ]
 
+# Frame Targets are not supported in WK1.
+# FIXME: <https://webkit.org/b/?> Add or rewrite tests to account for lack of frame targets in WK1.
```
Oops
```suggestion
# FIXME: <https://webkit.org/b/299207> Add or rewrite tests to account for lack of frame targets in WK1.
```

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/8c906311750eb5cf304ba3acc4d75c7bd5c705b7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/50623

the-chenergy committed:
Web Inspector: Introduce the Frame target type to prep for site isolation
rdar://143599045
bugs.webkit.org/show_bug.cgi?id=298742

Reviewed by BJ Burg.

This patch introduces the frame target type for Web Inspector. A target
represents a object that the inspector inspects, each carrying its own
resources and providing agents, groups of commands and events by domains
of functionalities. For a long time, there exists one page target per
web page, which worked fine with a single web process per page. Now with
site isolation being introduced, a better design is to treat each
individual frame in the page as an inspection unit, so we're not relying
on them all sharing the same web process to work.

The frame target is intended to eventually be the contact point for most
Web Inspector domains that were originally owned by the page target.
This patch is a step in that direction by introducing the frame target
type and merely making it possible to co-exist with other targets in
the frontend. Apart from existing, the frame target currently has no
domains (not even a Target domain) and relies on its page's Target
domain to notify its lifecycle events. The plan is to change that while
bringing in other domains in future commits.

The contents of this patch break down into three interlinked components:
1. Define the frame-related objects: FrameInspectorController, WebFrameInspectorTarget,
   and WebFrameInspectorTargetProxy in the backend, and FrameTarget in
   the frontend.
2. Relax parts of the frontend code to work with FrameTarget that has
   missing agents/domains, while updating relevant tests.
3. Add a test case for frames' lifecycle events linked to the creation
   and deletion of frame targets, also as a prototype of inspector tests
   with site isolation.

To summarize the structure of the frame-related objects, in comparison
with the existing page-related objects:
* In WebCore:
  - (old) Page owns InspectorController, which owns the agents
  - (new) Frame owns FrameInspectorController, which will own agents
    that we'll slowly create or move from the page
* In WebKit, web process:
  - (old) WebPage owns WebPageInspectorTarget through WebPageInspectorTargetController
  - (new) WebFrame owns WebFrameInspectorTarget directly, given we don't
    think there will be a need for a controller with just one target
    per frame
* In WebKit, UI process:
  - (old) WebPageProxy owns InspectorTargetProxy through WebPageInspectorController,
    which owns the Target agent
  - (new) We rename InspectorTargetProxy to WebPageInspectorTargetProxy
    to more easily introduce a frame-equivalent
  - (new) WebFrameProxy reuses the WebPageInspectorController to manage
    WebFrameInspectorTargetProxy

Test: http/tests/site-isolation/inspector/target/target.html
* LayoutTests/http/tests/inspector/target/provisional-load-cancels-previous-load.html:
* LayoutTests/http/tests/inspector/target/target-events-for-provisional-page.html:
* LayoutTests/http/tests/site-isolation/inspector/target/resources/leaf-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/inspector/target/resources/middle-iframe.html: Added.
* LayoutTests/http/tests/site-isolation/inspector/target/target-expected.txt: Added.
* LayoutTests/http/tests/site-isolation/inspector/target/target.html: Added.
* LayoutTests/inspector/unit-tests/target-manager-expected.txt:
* LayoutTests/inspector/unit-tests/target-manager.html:
* LayoutTests/inspector/worker/debugger-shared-breakpoint.html:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac-wk1/TestExpectations:
* Source/JavaScriptCore/inspector/InspectorTarget.h:
* Source/JavaScriptCore/inspector/agents/InspectorTargetAgent.cpp:
(Inspector::targetTypeToProtocolType):
* Source/JavaScriptCore/inspector/protocol/Target.json:
* Source/JavaScriptCore/inspector/scripts/codegen/models.py:
(validate_target_types):
* Source/WebCore/Headers.cmake:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/inspector/FrameInspectorController.cpp: Added.
(WebCore::FrameInspectorController::FrameInspectorController):
(WebCore::FrameInspectorController::~FrameInspectorController):
(WebCore::FrameInspectorController::ref const):
(WebCore::FrameInspectorController::deref const):
(WebCore::FrameInspectorController::createLazyAgents):
(WebCore::FrameInspectorController::connectFrontend):
(WebCore::FrameInspectorController::disconnectFrontend):
(WebCore::FrameInspectorController::dispatchMessageFromFrontend):
(WebCore::FrameInspectorController::developerExtrasEnabled const):
(WebCore::FrameInspectorController::canAccessInspectedScriptState const):
(WebCore::FrameInspectorController::functionCallHandler const):
(WebCore::FrameInspectorController::evaluateHandler const):
(WebCore::FrameInspectorController::frontendInitialized):
(WebCore::FrameInspectorController::executionStopwatch const):
(WebCore::FrameInspectorController::debugger):
(WebCore::FrameInspectorController::vm):
* Source/WebCore/inspector/FrameInspectorController.h: Added.
* Source/WebCore/page/Frame.cpp:
(WebCore::Frame::Frame):
(WebCore::Frame::protectedInspectorController):
* Source/WebCore/page/Frame.h:
(WebCore::Frame::inspectorController):
* Source/WebInspectorUI/Localizations/en.lproj/localizedStrings.js:
* Source/WebInspectorUI/UserInterface/Base/Main.js:
* Source/WebInspectorUI/UserInterface/Base/TargetType.js:
* Source/WebInspectorUI/UserInterface/Controllers/CanvasManager.js:
(WI.CanvasManager.prototype.setRecordingAutoCaptureFrameCount):
* Source/WebInspectorUI/UserInterface/Controllers/ConsoleManager.js:
(WI.ConsoleManager.prototype.initializeTarget):
(WI.ConsoleManager.prototype.requestClearMessages):
* Source/WebInspectorUI/UserInterface/Controllers/DOMDebuggerManager.js:
(WI.DOMDebuggerManager.prototype.addEventBreakpoint):
(WI.DOMDebuggerManager.prototype.addURLBreakpoint):
(WI.DOMDebuggerManager.prototype._handleEventBreakpointDisabledStateChanged):
(WI.DOMDebuggerManager.prototype._handleEventBreakpointEditablePropertyChanged):
(WI.DOMDebuggerManager.prototype._handleURLBreakpointDisabledStateChanged):
(WI.DOMDebuggerManager.prototype._handleURLBreakpointEditablePropertyChanged):
(WI.DOMDebuggerManager.prototype.allSupportedTargets):
(WI.DOMDebuggerManager):
* Source/WebInspectorUI/UserInterface/Controllers/DOMManager.js:
(WI.DOMManager.prototype._handleEventBreakpointEditablePropertyChanged):
* Source/WebInspectorUI/UserInterface/Controllers/DebuggerManager.js:
(WI.DebuggerManager.prototype.async initializeTarget):
(WI.DebuggerManager.prototype.set breakpointsEnabled):
(WI.DebuggerManager.prototype.setShouldBlackboxScript):
(WI.DebuggerManager.prototype.setShouldBlackboxPattern):
(WI.DebuggerManager.prototype.set asyncStackTraceDepth):
(WI.DebuggerManager.prototype.addSymbolicBreakpoint):
(WI.DebuggerManager.prototype._setBreakpoint):
(WI.DebuggerManager.prototype._removeBreakpoint):
(WI.DebuggerManager.prototype._updateSpecialBreakpoint):
(WI.DebuggerManager.prototype._handleSymbolicBreakpointDisabledStateChanged):
(WI.DebuggerManager.prototype._handleSymbolicBreakpointEditablePropertyChanged):
(WI.DebuggerManager.prototype._handleBlackboxBreakpointEvaluationsChange):
(WI.DebuggerManager.prototype._handleEngineeringPauseForInternalScriptsSettingChanged):
(WI.DebuggerManager.prototype._handleSourceCodeSourceMapAdded):
(WI.DebuggerManager.prototype.allSupportedTargets):
(WI.DebuggerManager):
* Source/WebInspectorUI/UserInterface/Controllers/NetworkManager.js:
(WI.NetworkManager.prototype.set emulatedCondition):
* Source/WebInspectorUI/UserInterface/Controllers/RuntimeManager.js:
(WI.RuntimeManager.prototype.initializeTarget):
* Source/WebInspectorUI/UserInterface/Controllers/TargetManager.js:
(WI.TargetManager.prototype._createTarget):
* Source/WebInspectorUI/UserInterface/Main.html:
* Source/WebInspectorUI/UserInterface/Models/HeapAllocationsInstrument.js:
(WI.HeapAllocationsInstrument.prototype.startInstrumentation):
(WI.HeapAllocationsInstrument.prototype.stopInstrumentation):
(WI.HeapAllocationsInstrument.prototype._takeHeapSnapshot):
(WI.HeapAllocationsInstrument.prototype.allSupportedTargets):
(WI.HeapAllocationsInstrument):
* Source/WebInspectorUI/UserInterface/Protocol/FrameTarget.js: Copied from Source/WebInspectorUI/UserInterface/Base/TargetType.js.
(WI.FrameTarget):
* Source/WebInspectorUI/UserInterface/Protocol/Target.js:
(WI.Target.prototype.initialize):
* Source/WebInspectorUI/UserInterface/Test.html:
* Source/WebInspectorUI/UserInterface/Views/HeapAllocationsTimelineView.js:
(WI.HeapAllocationsTimelineView.prototype._takeHeapSnapshotClicked):
* Source/WebInspectorUI/UserInterface/Views/LogContentView.js:
(WI.LogContentView.prototype._garbageCollect):
* Source/WebInspectorUI/UserInterface/Views/ScopeChainDetailsSidebarPanel.js:
* Source/WebInspectorUI/UserInterface/Views/SettingsTabContentView.js:
(WI.SettingsTabContentView.prototype._createConsoleSettingsView):
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.cpp:
(WebKit::InspectorTargetProxy::InspectorTargetProxy):
(WebKit::InspectorTargetProxy::create): Deleted.
(WebKit::InspectorTargetProxy::connect): Deleted.
(WebKit::InspectorTargetProxy::disconnect): Deleted.
(WebKit::InspectorTargetProxy::sendMessageToTargetBackend): Deleted.
(WebKit::InspectorTargetProxy::didCommitProvisionalTarget): Deleted.
(WebKit::InspectorTargetProxy::isProvisional const): Deleted.
* Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h:
* Source/WebKit/UIProcess/Inspector/WebFrameInspectorTargetProxy.cpp: Added.
(WebKit::WebFrameInspectorTargetProxy::create):
(WebKit::WebFrameInspectorTargetProxy::WebFrameInspectorTargetProxy):
(WebKit::WebFrameInspectorTargetProxy::connect):
(WebKit::WebFrameInspectorTargetProxy::disconnect):
(WebKit::WebFrameInspectorTargetProxy::sendMessageToTargetBackend):
(WebKit::WebFrameInspectorTargetProxy::didCommitProvisionalTarget):
(WebKit::WebFrameInspectorTargetProxy::isProvisional const):
* Source/WebKit/UIProcess/Inspector/WebFrameInspectorTargetProxy.h: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h.
(isType):
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.cpp:
(WebKit::WebPageInspectorController::init):
(WebKit::WebPageInspectorController::createWebPageInspectorTarget):
(WebKit::WebPageInspectorController::createWebFrameInspectorTarget):
(WebKit::WebPageInspectorController::didCreateProvisionalPage):
(WebKit::WebPageInspectorController::createInspectorTarget): Deleted.
* Source/WebKit/UIProcess/Inspector/WebPageInspectorController.h:
* Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.cpp: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.cpp.
(WebKit::WebPageInspectorTargetProxy::create):
(WebKit::WebPageInspectorTargetProxy::WebPageInspectorTargetProxy):
(WebKit::WebPageInspectorTargetProxy::connect):
(WebKit::WebPageInspectorTargetProxy::disconnect):
(WebKit::WebPageInspectorTargetProxy::sendMessageToTargetBackend):
(WebKit::WebPageInspectorTargetProxy::didCommitProvisionalTarget):
(WebKit::WebPageInspectorTargetProxy::isProvisional const):
* Source/WebKit/UIProcess/Inspector/WebPageInspectorTargetProxy.h: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h.
(isType):
* Source/WebKit/UIProcess/ProvisionalFrameProxy.cpp:
(WebKit::ProvisionalFrameProxy::protectedFrame const):
* Source/WebKit/UIProcess/ProvisionalFrameProxy.h:
(WebKit::ProvisionalFrameProxy::frame const):
* Source/WebKit/UIProcess/WebAuthentication/Mock/MockHidService.cpp:
* Source/WebKit/UIProcess/WebFrameProxy.cpp:
(WebKit::WebFrameProxy::WebFrameProxy):
(WebKit::WebFrameProxy::~WebFrameProxy):
(WebKit::WebFrameProxy::takeProvisionalFrame):
(WebKit::WebFrameProxy::provisionalLoadProcess):
(WebKit::WebFrameProxy::prepareForProvisionalLoadInProcess):
(WebKit::WebFrameProxy::sendMessageToInspectorFrontend):
* Source/WebKit/UIProcess/WebFrameProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::~WebPageProxy):
(WebKit::WebPageProxy::setFindClient):
(WebKit::WebPageProxy::launchProcess):
(WebKit::WebPageProxy::launchProcessForReload):
(WebKit::WebPageProxy::reload):
(WebKit::WebPageProxy::createInspectorTarget):
(WebKit::WebPageProxy::viewWillStartLiveResize):
(WebKit::WebPageProxy::wheelEventHandlingCompleted):
(WebKit::WebPageProxy::receivedPolicyDecision):
(WebKit::WebPageProxy::listenForLayoutMilestones):
(WebKit::WebPageProxy::didFailProvisionalLoadForFrame):
(WebKit::WebPageProxy::didReceiveTitleForFrame):
(WebKit::WebPageProxy::nonEphemeralWebPageProxy):
(WebKit::WebPageProxy::logFrameNavigation):
(WebKit::WebPageProxy::decidePolicyForNewWindowAction):
(WebKit::WebPageProxy::didPerformClientRedirectShared):
(WebKit::WebPageProxy::didPerformServerRedirectShared):
(WebKit::WebPageProxy::setMediaVolume):
(WebKit::WebPageProxy::Internals::currentlyProcessedMouseDownEvent):
(WebKit::WebPageProxy::contextMenuItemSelected):
(WebKit::WebPageProxy::updateEditorState):
(WebKit::WebPageProxy::processDidBecomeResponsive):
(WebKit::WebPageProxy::setScrollPinningBehavior):
(WebKit::WebPageProxy::resetSpeechSynthesizer):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTarget.cpp: Added.
(WebKit::WebFrameInspectorTarget::WebFrameInspectorTarget):
(WebKit::WebFrameInspectorTarget::protectedFrame):
(WebKit::WebFrameInspectorTarget::identifier const):
(WebKit::WebFrameInspectorTarget::connect):
(WebKit::WebFrameInspectorTarget::disconnect):
(WebKit::WebFrameInspectorTarget::sendMessageToTargetBackend):
(WebKit::WebFrameInspectorTarget::toTargetID):
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTarget.h: Copied from Source/WebKit/UIProcess/Inspector/InspectorTargetProxy.h.
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTargetFrontendChannel.cpp: Copied from Source/WebKit/UIProcess/ProvisionalFrameProxy.h.
(WebKit::WebFrameInspectorTargetFrontendChannel::WebFrameInspectorTargetFrontendChannel):
(WebKit::WebFrameInspectorTargetFrontendChannel::protectedFrame):
(WebKit::WebFrameInspectorTargetFrontendChannel::sendMessageToFrontend):
* Source/WebKit/WebProcess/Inspector/WebFrameInspectorTargetFrontendChannel.h: Copied from Source/WebKit/UIProcess/ProvisionalFrameProxy.h.
* Source/WebKit/WebProcess/UserContent/WebUserContentController.cpp:
* Source/WebKit/WebProcess/WebCoreSupport/WebLocalFrameLoaderClient.cpp:
(WebKit::WebLocalFrameLoaderClient::dispatchDidCommitLoad):
* Source/WebKit/WebProcess/WebCoreSupport/WebPermissionController.cpp:
(WebKit::WebPermissionController::query):
* Source/WebKit/WebProcess/WebPage/WebFrame.cpp:
(WebKit::WebFrame::WebFrame):
(WebKit::WebFrame::connectInspector):
(WebKit::WebFrame::disconnectInspector):
(WebKit::WebFrame::sendMessageToInspectorTarget):
* Source/WebKit/WebProcess/WebPage/WebFrame.h:
* Source/WebKit/WebProcess/WebPage/WebFrame.messages.in:

Canonical link: https://commits.webkit.org/300338@main

webkit-commit-queue commented:
Committed 300338@main (06f8ad1a5a66): <https://commits.webkit.org/300338@main>

Reviewed commits have been landed. Closing PR #50623 and removing active labels.
